configs/next/packages/python/stage_optimized

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+#
+# Copyright 2021 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Python build parameters for stage optimized
+# ===========================================
+#
+
+# Tell the build system to hold the temp install folder
+ATCFG_HOLD_TEMP_INSTALL='no'
+# Tell the build system to hold the temp build folder
+ATCFG_HOLD_TEMP_BUILD='no'
+# Build in a new directory
+ATCFG_BUILD_STAGE_T='dir'
+# Don't fail if stage final install place doesn't exist
+ATCFG_INSTALL_PEDANTIC=no
+
+current_processor=$(awk --field-separator=: '/^cpu/ {print $2; exit}' /proc/cpuinfo | sed 's/^.*POWER\([[:digit:]]\+\).*$/\1/')
+target_processor=$(echo ${AT_OPTIMIZE_CPU} | sed 's/^.*power\([[:digit:]]\+\).*$/\1/')
+
+atcfg_configure() {
+	if [[ "${cross_build}" == 'no' && "$current_processor" -ge "$target_processor" ]]; then
+		PATH=${at_dest}/bin:${PATH} \
+		CC="${at_dest}/bin/gcc -m${compiler}" \
+		CXX="${at_dest}/bin/g++ -m${compiler}" \
+		CFLAGS="-g -Wformat" \
+		CXXFLAGS="-g" \
+		OPT="-O3 -mcpu=${AT_OPTIMIZE_CPU/ppc/}" \
+		${ATSRC_PACKAGE_WORK}/configure \
+			--build=${target} \
+			--host=${target} \
+			--target=${target} \
+			--prefix=${at_dest} \
+			--exec-prefix=${at_dest} \
+			--libdir="${at_dest}/lib${compiler##32}" \
+			--with-platlibdir="lib${compiler##32}" \
+			--enable-shared \
+			--enable-loadable-sqlite-extensions \
+			--with-openssl="${at_dest}" \
+			--with-ssl-default-suites=openssl \
+			--enable-optimizations \
```

Comment:
This is actually not the issue. Since the build itself runs the built `python` binary, any target-optimized build will fail unless it is being built on the target.
If we want target-optimized Python binaries, with or without profile-guided optimization, the build must happen on that target.

Reply:
We can't build cpu-tuned libraries if they require the latest processor.
We really need to find a way to build libpython without running new instructions, even if it is just a workaround.

Reply:
Just for clarity, the changes here will not run unsupported instructions on any processor. The build self-limits to what can be supported on the build system.  The downside, of course, is that if you build on Power9, for example, you won't get Power10 optimizations.
Let me probe on this a bit:
> We can't build cpu-tuned libraries if they require the latest processor.
- Is the goal to build with a GCC compiler that supports processorB with `-mcpu=B` on a system running earlier processorA? It sounds like "yes".
- Do we abandon _all_ profile-guided optimization? The only reasonable profile-guided optimization must be done on the latest _available_ (not just supported by GCC) processor.  If you try to do profile-guided optimization on an earlier processor, you may not get useful optimizations for a current processor. It sounds like this is "yes", too. I don't know what we lose by doing this.

So, I'll work to find a way to build Power9 and Power10 optimized libpython.so on Power8.

Reply:
> Is the goal to build with a GCC compiler that supports processorB with `-mcpu=B` on a system running earlier processorA? It sounds like "yes".

Yes.

> Do we abandon all profile-guided optimization?

I also think this is correct.

> I don't know what we lose by doing this.

OK.

> So, I'll work to find a way to build Power9 and Power10 optimized libpython.so on Power8.

After removing the `--enable-optimizations` and restricting the build to `libpython3.so`, this will work.

Reply:
> > So, I'll work to find a way to build Power9 and Power10 optimized libpython.so on Power8.
> 
> After removing the `--enable-optimizations` and restricting the build to `libpython3.so`, this will work.

Indeed! It was easier than I was anticipating. I didn't realize there was a `libpython3.so` target that didn't need (or run) the `python` executable.

Thanks, @tuliom ! I'll push new changes here imminently.